File: services/client/src/packet/parser.go
package packet

import "errors"

const HEADER_LEN int = 1

type Deserializer interface {
	Deserialize([]byte, uint32) (Packet, error)
}

type Parser struct{}

func (p Parser) Deserialize(bytes []byte, agencyId uint32) (Packet, error) {
	if len(bytes) < HEADER_LEN {
		return nil, errors.New("Invalid header length")
	}
	switch bytes[0] {
	case TYPE_BET:
		return BetInfoFromBytes(bytes, agencyId)
	case TYPE_END:
		return EndFromBytes(bytes)
	default:
		return nil, errors.New("Unknown packet type")
	}
}

File: services/server/src/packet/parser.py
from packet.packet import Packet, TYPE_BET, TYPE_END
from packet.bet_info_packet import bet_info_from_bytes
from packet.end_packet import end_from_bytes

HEADER_LEN = 1

class Deserializer:
    def deserialize(self, bytes: bytes, agency_id: int) -> Packet:
        if len(bytes) < HEADER_LEN:
            raise ValueError("Invalid header length")
        if bytes[0] == TYPE_BET:
            return bet_info_from_bytes(bytes, agency_id)
        if bytes[0] ==  TYPE_END:
            return end_from_bytes(bytes)
        raise ValueError("Unknown packet type")
